chain-signatures/node/src/protocol/cryptography.rs

```diff
@@ -420,6 +426,13 @@ impl CryptographicProtocol for RunningState {
         // block height is up to date, such that they too can process signature requests. If they cannot
         // then they are considered unstable and should not be a part of signature generation this round.
         let stable = ctx.mesh().stable_participants().await;
+        if stable.len() < self.threshold {
+            tracing::info!("Require at least {} stable participants, got {}: {:?}", self.threshold, stable.len(), stable.keys_vec());
+            drop(messages);
+            drop(presignature_manager);
+            return Ok(NodeState::Running(self));
+        }
```

Comment:
we shouldn't be early returning in this function because the rest of the code then doesn't take effect at all. We should just be doing this early check in the `organize` and `handle_requests` function where `stable` is actually being used

Reply:
I was thinking skip the following step is fine, as protocol cannot proceed without at least threshold stable participant any way, and logs show that it will retry every 1s until there's >= threshold stable participant, and this step will proceed, wdyt?

Reply:
So there's the messages that doesn't get propagated if we don't early return which is need for triples and presignatures

Reply:
The logic after `let stable = ...` is:
```
sign_queue.organize(stable) (-> my_requests)
signature_manager.handle(my_requests, stable)
messages.push(signature_manager.poke())
messages.send_encrypted
```
My understanding is they are all depending on stable participants. Correct me if I'm wrong

Reply:
`messages.send_encrypted` will send more than just signature messages. Look at all the `messages.push` before we get to signatures

Reply:
Ah got it

Reply:
Done